### ISSUE TITLE
[Issue]: Audio track reset on next episode

### ISSUE THREAD

**Issue by Plinz** [jellyfin/jellyfin-web#3994]
Please describe your bug

When I'm watching a TV show with an audio track that is not the default one, and I finish an episode or use button next episode, the next episode doesn't keep the previous audio track on. But, on UI, it shows that it is selected, so I need to set another one first and then set back the one I want.

Step to reproduce:

    watch a TV show with multiple audio track
    choose a different audio track than the default one
    go to next episode

Expected:

    The audio track is kept in the next episode.

What happen instead:

    The audio track from previous episode is selected, but default audio track is playing.

Jellyfin Version
10.8.4

    OS: Linux
    Virtualization: no
    Clients: Browser
    Browser: Firefox 105.0.1
    FFmpeg Version: 5.1.1
    Playback Method: Streaming direct
    Hardware Acceleration: none
    Plugins: none
    Reverse Proxy: nginx
    Base URL: /jellyfin
    Networking: Host
    Storage: local

**Comment by kiawizard** [comment]
This is a very important issue, please do not close it. Imagine you have a series with 2 different language audio tracks. You are watching the secondary language now. If you have "Play default audio track regardless of language" unchecked in your profile (default), at the beginning of each episode you have to switch audio track twice - to 1st, then to 2nd. With the setting checked, you have to change audio once per episode. Normally you'd want to select 2nd language once and then watch the whole season :-)

**Comment by kiawizard** [comment]
this issue still affects you in the latest release

**Comment by jmerdich** [comment]
Still present on 10.8.11 (tested on Firefox, Chrome, and Android-app-webplayer).

This wouldn't be so bad if the default audio track was playing **and** the audio dropdown indicated it, but because they're mismatched, you have to change audio settings twice to get to the language that was playing on the previous episode.

**Comment by heschlie** [comment]
Seeing the same issue at least on the web client, I haven't seen it on the Android client. Also on Firefox, haven't tested on other browsers.

**Comment by vladkar** [comment]
I’m experiencing the same issue where I need to reload the audio track for each episode. Although it appears correct in the user interface, I have to switch it back to a different audio track and then select the desired one. If I use subtitles, I also need to turn them off and then back on again because they freeze after the first adjustment. As a result, I have to make four changes in total for each episode if I need both an audio track and subtitles. This issue has persisted for years across various versions.

**Comment by filipfigzalski** [comment]
I have the same issue, after each episode language changes to the default one, even though the GUI shows otherwise.

**Comment by tipyn** [comment]
Same issue here. It’s pretty annoying as I have quite a few users on my server and it makes them lose trust in the app and not want to use it.

**Comment by ntnkrsh** [comment]
Same issue here... Took time to notice it because have watched 1 episode at a time.
Hope it will get solved

**Comment by KraXen72** [comment]
This issue is still present, details:
```
Playback Info
Player
Html Video Player
Play method
Transcoding
Protocol
https
Stream type
HLS
Video Info
Player dimensions
2560x1335
Video resolution
1920x1080
Dropped frames
5
Corrupted frames
0
Transcoding Info
Video codec
H264
Audio codec
AAC
Audio channels
2
Bitrate
6.1 Mbps
Transcoding progress
77.5%
Transcoding framerate
531 fps
Reason for transcoding
The video codec is not supported
Secondary audio tracks are not supported
Original Media Info
Container
webm
Size
260.5 MiB
Bitrate
1.6 Mbps
Video codec
HEVC Main 10
Video bitrate
1.4 Mbps
Video range type
SDR
Audio codec
OPUS
Audio bitrate
106 kbps
Audio channels
2
Audio sample rate
48000 Hz
```

**Comment by ConnorS1110** [comment]
I am experiencing the same thing still that is mentioned in the original post. With the `Set audio/subtitle track based on previous item` turned on, it is selecting the correct audio/subtitle track, but it is actually using whatever the default was instead. Web version 10.10.

**Comment by ebkalderon** [comment]
Perhaps this issue should be renamed from "Audio track" to "Audio/Subtitle tracks" to reflect the actual current state of discussion? This bug has affected both audio and subtitle tracks for ~2 years now, and it was tricky for me to locate this ticket via GH search due to the misleading name.

For the record: This bug still occurs on Jellyfin Web 10.9.X in both Firefox and Chrome. It's _very_ irritating when I'm trying to binge a TV show with friends and family and I have to keep pausing, switching subtitles off and then back on again, and resuming playback every time Jellyfin advances to the next episode.

**Comment by vegiemane** [comment]
Any way I can bump this? This is probably the only thing I don't like about Jellyfin as a fellow anime watcher or when watching a TV show that has different language and audio tracks.

The same thing seems to happen, you select the correct audio or subtitle track you'd like to watch with, works fine, you go to the next episode with the `Set audio/subtitle track based on previous item` option selected, it says you have the correct audio track or subtitle track selected on the front end but you can hear the default audio track being played or you can view the default subtitle track being played.

The only workaround I have for this is to download the files to my local machine and change the default audio or subtitle track default flags then re-upload to my Jellyfin box - particularly annoying when you have a hardlink setup with something like the *arr apps or vice versa.

**Comment by friesoft** [comment]
This doesn't seem to be fixed. We are still having the same problem (audio track and subtitles resetting on next episode but showing correctly in the menu) on current versions (Android TV app (0.18.11), Android app (2.6.3) and in the web (10.10.7))

There are also other posts that this issue still exists, like: https://github.com/jellyfin/Swiftfin/issues/771#issuecomment-3349319831

**Comment by viown** [comment]
@friesoft Do you have "Set audio/subtitle track based on previous item" enabled in Settings -> Playback?

**Comment by friesoft** [comment]
Yes I have both flags for audio and subtitle enabled. Default language for audio is set as well.

**Comment by viown** [comment]
That's odd as the issue you described should be fixed. If you just recently updated Jellyfin, there is an issue with the android app where it caches an older version of web. Can you see if you can still reproduce this in the browser with cache cleared?

**Comment by friesoft** [comment]
I'm new to jellyfin so all installs (server/web/apps) are fresh with the latest versions.
